synonymdev/bitkit-react-native
# refactor(wallet): Implement Rust Address Generator

### Description
- Removes `nodejs-mobile` dependency and related files.
- Adds `react-native-address-generator` and implements it accordingly.
- The `BitcoinActions` class was moved from `nodejs-assets/nodejs-project` to `utils/bitcoin-actions` and updated accordingly.
- This update brings with it a significant performance improvement when generating addresses.
- If ya'll agree that this way of generating addresses in Bitkit is an improvement over the previous `nodejs-mobile` implementation and approve the PR we'll just need to import/move the `react-native-address-generator` library to the `synonymdev` name and make it public to continue working on from there moving forward. 

### Linked Issues/Tasks
- #1617

### Type of change

### Tests

### QA Notes
- Please ensure address generation still works as expected on iOS & Android for each address type and network.
